Refactor store and wallet API to add multichain support

matiasgali opened the pull request.

matiasgali committed:
Extract constants from "walletSlice.js".

matiasgali committed:
Refactor reducers.

batudo commented on src/helpers/wallet/index.js:
```diff
@@ -0,0 +1,48 @@
+
+  const { modal, provider } = connection
+
+  if (provider?.close)
+    await provider.close()
+
+  if (modal.clearCachedProvider)
+    await modal.clearCachedProvider()
```
I personally prefer this style, but since on some other places it is used with curly braces, let's keep it that way for the time being.
```suggestion
  if (provider?.close) {
    await provider.close()
  }

  if (modal.clearCachedProvider) {
    await modal.clearCachedProvider()
  }
```

batudo commented on src/hooks/wallet.js:
```diff
@@ -0,0 +1,38 @@
+    const onChainChanged = chainId => dispatch(chainChanged({ chainId }))
+
+    const onAccountsChanged = () => {
+      const address = provider.selectedAddress || undefined
```
```suggestion
      const address = provider.selectedAddress || undefined

```

batudo commented on src/features/vaultsSlice.js:
```diff
@@ -18,9 +22,14 @@ const initialState = {
 
 const selectOrder  = state => state.vaults.order
 const selectErrors = state => state.vaults.errors
+export const selectStatus = state => state.vaults.status
```
```suggestion

export const selectStatus = state => state.vaults.status
```

batudo reviewed (approved):
All suggestions are just _little_ style fixes, so I'll apply them and merge right away 👍

batudo commented:
I've "aborted" the merge because it breaks the build:
<img width="1239" alt="Captura de Pantalla 2021-10-03 a la(s) 10 17 18" src="https://user-images.githubusercontent.com/82539708/135755366-f4081293-84e3-42c9-902a-1444d794fe4e.png">

Let's hope it's just a _quick_ fix =)

matiasgali committed:
Refactor internal wallet API.

- Decouple wallet methods from Redux.
- Add a minimal surface API to interact with wallet.
